src/command/defaults/ClearCommand.php

```diff
@@ -0,0 +1,174 @@
+use pocketmine\command\CommandSender;
+use pocketmine\command\utils\InvalidCommandSyntaxException;
+use pocketmine\item\LegacyStringToItemParser;
+use pocketmine\lang\TranslationContainer;
+use pocketmine\player\Player;
+use pocketmine\utils\TextFormat;
+use function array_merge;
+use function count;
+
+class ClearCommand extends VanillaCommand{
+
+	public function __construct(string $name){
+		parent::__construct(
+			$name,
+			"%pocketmine.command.clear.description",
+			"%pocketmine.command.clear.usage"
+		);
+		$this->setPermission("pocketmine.command.clear.self;pocketmine.command.clear.other");
+	}
+
+	public function execute(CommandSender $sender, string $commandLabel, array $args){
+		if(!$this->testPermission($sender)){
+			return true;
+		}
+
+		if(count($args) > 3){
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$target = null;
+		if(isset($args[0])){
+			if(!$sender->hasPermission("pocketmine.command.clear.other")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender->getServer()->getPlayer($args[0]);
+			if($target === null){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.player.notFound"));
+				return true;
+			}
+		}elseif($sender instanceof Player){
+			if(!$sender->hasPermission("pocketmine.command.clear.self")){
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender;
+		}else{
+			throw new InvalidCommandSyntaxException();
+		}
+
+		$item = null;
+		$maxCount = -1;
+		if(isset($args[1])){
+			try{
+				$item = LegacyStringToItemParser::getInstance()->parse($args[1]);
+
+				if(isset($args[2])){
+					$item->setCount($maxCount = $this->getInteger($sender, $args[2], 0));
+				}
+			}catch(InvalidArgumentException $e){
+				//vanilla checks this at argument parsing layer, can't come up with a better alternative
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.give.item.notFound", [$args[1]]));
+				return true;
+			}
+		}
+
+		//checking players inventory for all the items matching the criteria
+		if($item !== null and $maxCount === 0){
+			$count = 0;
+			$contents = array_merge($target->getInventory()->all($item), $target->getArmorInventory()->all($item));
+			foreach($contents as $content){
+				$count += $content->getCount();
+			}
+
+			if($count > 0){
+				Command::broadcastCommandMessage($sender, new TranslationContainer("%commands.clear.testing", [$target->getName(), $count]));
+			}else{
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.clear.failure.no.items", [$target->getName()]));
+			}
+
+			return true;
+		}
+
+		$cleared = 0;
+
+		//clear everything from the targets inventory
+		if($item === null){
+			$contents = array_merge($target->getInventory()->getContents(), $target->getArmorInventory()->getContents());
```

Comment:
Count of cursor inventory is missing here.

Reply:
I'm not sure if `clear` is intended to clear the cursor or not.

Reply:
Does the count of cursor inventory matter tho?

Reply:
Yes, I would think so. Either don't clear it, or include it in the total.

Reply:
I opted, not to clear it.